### PR TITLE
Introduction of the variable prodlnox for EMAC

### PR DESCRIPTION
**Pull request by FranziskaWinterstein**
The changes introduce the variable prodlnox as the integrated sum of NOxcg and NOxic divided by the timestep length.

## Description

The variable is introduced by a fix in emac fixes. It requires NOxcg, NOxic and dt being available in the given channel.

prodlnox will be on dimension `[time]` only.  It is derived as kg s-1 (conversion to the displayed Tg yr-1 per preprocessor possible).

Result looks like this:

timeseries_EMAC_prodlnox

***

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

### CODE REVIEW

File: esmvalcore/cmor/_fixes/emac/emac.py [hunk 1]
```diff
@@ -211,6 +211,30 @@ def fix_metadata(self, cubes):
...
+                                               weights=None) +
+            self.get_cube(cubes, var_name=['NOxic_cav', 'NOxic_ave',
+                                           'NOxic']).collapsed(
+                                               ['longitude', 'latitude'],
+                                               iris.analysis.SUM,
+                                               weights=None)) /
+                self.get_cube(cubes, var_name=['dt'])) * 365 * 24 * 3600 / 1e9
+
+        cube.var_name = self.vardef.short_name
+        cube.units = 'Tg a-1'
```
Comment (schlunma) [inline esmvalcore/cmor/_fixes/emac/emac.py:233]:
```suggestion
                self.get_cube(cubes, var_name=['dt'])
        )
        cube.convert_units('Tg yr-1')
        cube.var_name = self.vardef.short_name
```

This should be safer and easier to read. Also note that the unit `a` is interpreted as are (100m2), not year.

File: esmvalcore/cmor/tables/custom/CMOR_prodlnox.dat [hunk 2]
```diff
@@ -0,0 +1,20 @@
+SOURCE: CMIP6
+!============
+variable_entry:    prodlnox
+!============
+modeling_realm:    atmos
+!----------------------------------
+! Variable attributes:
+!----------------------------------
+standard_name:
+units:             Tg a-1
```
Comment (schlunma) [inline esmvalcore/cmor/tables/custom/CMOR_prodlnox.dat:10]:
```suggestion
units:             Tg yr-1
```

File: esmvalcore/cmor/_fixes/emac/emac.py [hunk 3]
```diff
@@ -211,6 +211,30 @@ def fix_metadata(self, cubes):
...
+    """Fixes for ``prodlnox``."""
+
+    def fix_metadata(self, cubes):
+        """Fix metadata."""
+        cube = ((
+            self.get_cube(cubes, var_name=['NOxcg_cav', 'NOxcg_ave',
+                                           'NOxcg']).collapsed(
+                                               ['longitude', 'latitude'],
+                                               iris.analysis.SUM,
+                                               weights=None) +
```
Comment (schlunma) [inline esmvalcore/cmor/_fixes/emac/emac.py:224]: Just to make sure I'm getting this right: this variable is simply the total mass of that quantity in the grid cell, that's why you don't need weights for the sum?

**Review by schlunma — COMMENTED** [review]
Looks very good @FranziskaWinterstein, just got a minor comment.

Would you be able to add some tests for this? I think you can mostly copy-paste [this code](https://github.com/ESMValGroup/ESMValCore/blob/0611f84a1882713b50d2eff10af3a0807f5d511d/tests/integration/cmor/_fixes/emac/test_emac.py#L907-L931) 👍

File: esmvalcore/cmor/_fixes/emac/emac.py (hunk 1 above)
Comment (FranziskaWinterstein) [reply esmvalcore/cmor/_fixes/emac/emac.py:233]: Thank you for this comment. I was not aware of that. Do you know, if the cf unit converter is able to convert kg s-1 to Tg yr-1? I think it makes more sense to define the variable in SI units and do the unit conversion by the preprocessor.

Comment (schlunma) [reply esmvalcore/cmor/_fixes/emac/emac.py:233]: Yes, it is (I just tested it). It's also fine for me to use the SI, but if you'd like you can keep it like that. We have lots of custom tables that do not use SI units.

Comment (FranziskaWinterstein) [reply esmvalcore/cmor/_fixes/emac/emac.py]: It seems like converting the units at that point does not work, as it assumes some strange untis for the cubes of NOxcg and NOxic. They are defined with `kg(N)` in MESSy and that might be a problem as `N` is assumed as Newton by the conversion.
I suppose I have to overwrite them.

**Review by schlunma — APPROVED** [review]
Great!

**Pull request merged**